### DISCUSSION TITLE
Show data sources of each panel within one dashboard.

### DISCUSSION THREAD

**Discussion by kevisong** [grafana/grafana#37187] [Dashboards, Panels & Visualizations]
When a dashboard contains multiple panels, where some panels have different data sources, it's difficult to filter out panel by specific data source.

It is possible to show (or toggle to show) data source of each panel in dashboard page?

**Comment by torkelo** [comment]
You can use a data source variable to change the datasource used by panel queries using a dashboard wide variable drop down

**Comment by Mario-Hofstaetter** [comment]
> It is possible to show (or toggle to show) data source of each panel in dashboard page?

While it is not possible to toggle this information (at least not without dirty hacks?), you may visualize it yourself in the panel title if necessary by using a datasource variables as mentioned by Torkel, although it is a bit cumbersome.

If the datasources are fixed for a given panel it may be easier to just add text like "from loki" / "from prometheus" ...

Using https://play.grafana.org/d/000000029/prometheus-demo-dashboard :

### Add datasource variable

![grafik](https://user-images.githubusercontent.com/33002073/127454653-6b9c7416-2776-45e2-8fc0-868a3b5efbdc.png)

### Use in panel

![grafik](https://user-images.githubusercontent.com/33002073/127454758-e0b7fe69-f218-4860-8987-7529b5691395.png)

### Place the variable e.g. in panel title

![grafik](https://user-images.githubusercontent.com/33002073/127454904-ed4a5053-269d-4d9e-99a4-003d6c49b1d2.png)

### Will show the used datasource (its name)

![vy1LX2JnaJ](https://user-images.githubusercontent.com/33002073/127455143-98ed6bc4-eba2-4e05-8fce-afbc65277de8.gif)

If the user must not change the datasource, you can just hide the datasource variable.

![grafik](https://user-images.githubusercontent.com/33002073/127455326-3b257191-1054-4183-9bfd-1826b2d118dd.png)

Then you need to ensure the right value is the first and only one.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
